technologydata/utils.py

```diff
@@ -82,3 +206,43 @@ def change_datetime_format(
             return output_datetime_string
         except ValueError as e:
             raise ValueError(f"Error during datetime formatting: {e}")
+
+    @staticmethod
+    def replace_special_characters(input_string: str) -> str:
+        """
+        Replace special characters and spaces in a string with underscores,
+        collapsing multiple consecutive underscores into a single underscore.
+        Finally, it lowercases all characters of the string and removes leading or
+        trailing underscores.
+
+        Parameters
+        ----------
+        input_string : str
+            The input string from which special characters and spaces will be replaced.
+
+        Returns
+        -------
+        str
+            A new string with all special characters and spaces replaced by a single underscore
+            where consecutive underscores occur.
+
+        Examples
+        --------
+        >>> replace_special_characters("Hello, World! Welcome to Python @ 2023.")
+        'Hello_World_Welcome_to_Python_2023_'
+
+        >>> replace_special_characters("Special#Characters$Are%Fun!")
+        'Special_Characters_Are_Fun_'
```

Comment:
```suggestion
        'Hello_World_Welcome_to_Python_2023'

        >>> replace_special_characters("Special#Characters$Are%Fun!")
        'Special_Characters_Are_Fun'
```

The trailing `_` is stripped, no?